File: apps/web/ce/types/project-copy.ts
/**
 * Copyright (c) 2023-present Plane Software, Inc. and contributors
 * SPDX-License-Identifier: AGPL-3.0-only
 * See the LICENSE file for details.
 */

export type TProjectCopyJobStatus = "queued" | "processing" | "completed" | "failed";

export interface IProjectCopyJob {
  job_id: string;
  status: TProjectCopyJobStatus;
  new_project_id: string | null;
  error: string | null;
}

File: apps/api/plane/app/serializers/project_copy.py
# Copyright (c) 2023-present Plane Software, Inc. and contributors
# SPDX-License-Identifier: AGPL-3.0-only
# See the LICENSE file for details.

from .base import BaseSerializer
from plane.db.models import ProjectCopyJob


class ProjectCopyJobSerializer(BaseSerializer):
    class Meta:
        model = ProjectCopyJob
        fields = [
            "id",
            "source_project_id",
            "target_workspace_id",
            "initiated_by_id",
            "identifier",
            "name_override",
            "status",
            "new_project_id",
            "error",
            "created_at",
        ]
        read_only_fields = ["id", "created_at"]
